GrammarNazi.Core/Utilities/TelegramUpdateHandler.cs

```diff
@@ -52,6 +54,16 @@ public Task HandleError(ITelegramBotClient botClient, Exception exception, Cance
                 return Task.CompletedTask;
             }
 
+            var innerExceptions = exception.GetInnerExceptions();
+
+            if (innerExceptions.Any(x => x.GetType() == typeof(SocketException) && x.Message.Contains("Connection reset by peer")))
```

Comment:
I just realized I could've used ... `x is SocketException` ... here

Reply:
yep, go ahead @nminaya